src/YHttpWidgetsActionHandler.cc

```diff
@@ -239,71 +229,143 @@ int YHttpWidgetsActionHandler::do_action(YWidget *widget, const std::string &act
         {
-            body << "Action is not supported for the selected widget: " << widget->widgetClass() << std::endl;
-            return MHD_HTTP_NOT_FOUND;
+            return action_handler<YTimeField>( widget, body, [&] (YTimeField *input) {
+                yuiMilestone() << "Setting value for YTimeField \"" << input->label() << '"' << std::endl;
+                input->setKeyboardFocus();
+                input->setValue( value );
+                activate_widget( input );
+            } );
         }
+        else if ( dynamic_cast<YComboBox*>(widget) )
+        {
+            YComboBox* cb = dynamic_cast<YComboBox*>(widget);
+            if( !cb->editable() )
+            {
+                body << "{ \"error\" : \"Combobox '" << cb->label() << "' is not editable \" }" << std::endl;
+                return MHD_HTTP_UNPROCESSABLE_ENTITY;
+            }
+
+            return action_handler<YComboBox>( widget, body, [&] (YComboBox *cb) {
+                yuiMilestone() << "Setting value for YComboBox \"" << cb->label() << '"' << std::endl;
+                cb->setKeyboardFocus();
+                cb->setValue( value );
+                activate_widget( cb );
+            } );
+        }
+
+        std::ostringstream error;
+        error << "Action 'enter_text' is not supported for the selected widget: \"" << widget->widgetClass() << "\"";
+        return handle_error( body, error.str(), MHD_HTTP_NOT_FOUND );
     }
     else if ( action == "select" )
     {
         std::string value;
         if (const char* val = MHD_lookup_connection_value(connection, MHD_GET_ARGUMENT_KIND, "value"))
             value = val;
-
         if ( dynamic_cast<YComboBox*>(widget) )
         {
-            return action_handler<YComboBox>(widget, [&] (YComboBox *cb) {
+            return action_handler<YComboBox>( widget, body, [&] (YComboBox *cb) {
                 yuiMilestone() << "Activating ComboBox \"" << cb->label() << '"' << std::endl;
                 cb->setKeyboardFocus();
-                cb->setValue(value);
+                YItem * item = cb->findItem(value);
+                if ( item )
+                {
+                    yuiMilestone() << "Activating Combobox \"" << cb->label() << '"' << std::endl;
+                    cb->selectItem( item );
+                    activate_widget( cb );
+                }
+                else
+                {
+                    throw YUIException("Item: '" + value + "' cannot be found in the combobox");
+                }
             } );
         }
         else if( dynamic_cast<YTable*>(widget) )
         {
+            int row_id = 0;
+            if ( const char* val = MHD_lookup_connection_value(connection, MHD_GET_ARGUMENT_KIND, "row") )
+                row_id = atoi(val);
+            // Handle case when want select row by row number
+            return action_handler<YTable>( widget, body, [&] (YTable *tb) {
+
+                if( row_id >= tb->itemsCount() ) {
```

Comment:
You can use [YTable::hasColumn()](https://github.com/libyui/libyui/blob/master/src/YTable.h#L95) directly.

Reply:
That would be useful for checking `column` parameter, but for the row we should check number of items. Or am I missing something?